MaeBlock.hpp

```diff
@@ -359,7 +360,7 @@ get_indexed_property(const std::map<std::string, std::shared_ptr<T>>& map,
 template <typename T>
 inline void set_indexed_property(std::map<std::string, std::shared_ptr<T>>& map,
                                  const std::string& name,
-                                 const std::shared_ptr<T> value)
+                                 std::shared_ptr<T> value)
 
 {
     auto iter = map.find(name);
```

Comment:
The body of this function seems like it is merely doing:

    map[name] = value;

am I missing something? I feel like I'd want to replace any calls to this with:

    map[name] = std::move(value);